Architecture.md

```diff
@@ -0,0 +1,112 @@
+
+## Value-Object
+
+Similar to `Entity`s, `Value-Object`s also abstract a _business domain object_. The core difference between `Entity`s and `Value-Object`s is that `Value-Object`s do not have a unique identifier.
+`Value-Object`s too are defined by their data shape and their business logic, and those too are defined through `Schema` and `Core` logic inthe same packages where `Entity`s are defined.
+
+An example of a `Value-Object` might be an email or a geographical `Address`. An `Order` might have a delivery `Address` but this `Address` is only defined by its content. `Value-Object`s are generally attached to `Entity`s and modifications of `Value-Object`s are indistinguishable from replacing the `Value-Object`s as they do not hold any unique identifier and their identity is defined by their content only.
+
+## Schema
+
+A `Schema` is a blueprint for generating:
+
+- encoders and decoders for the `Model`. The `Model` defines the data shape of the _business domain object_
+- `Api`s. Utilities to derive other `Schema`s from a base `Model`
+- interpreters like `arbitraty` (functions able to generate fake data based on the `Model` definition.)
+
+Should be noted that `Schema`s can encode more than basic data type information such as specific constraints. E.g. a `DateString` can encode not only that the underlying data is a `string` but whether _any_ given `string` can be encoded or decoded to valid `Date` object representation.
+
+## UseCase
+
+`UseCase`s describe the specific application-layer capabilities.
+
+Each capacity is implemented through a single `Controller`.
+
+An example might be that two different applications may both interact with `Order`s but they may not necessarily have the same capabilities. A user-facing application, such as our usual ecommerce example, may give the capabilities to create and review `Order`s and `Item`s.
+
+Another application used in a backoffice might also interact with `Order`s but witha different scope such as handling updates, communication and payments but it may not create or review orders.
+
+Another application used in a warehouse may handle the inventory, packaging and operations with the carrier. But it may neither have the capabilities to create or review orders, nor to handle the payments or communication with the customer.
+
+## Controller
+
+A `Controller` is a piece of application-specific logic. A `Conteoller` defines a single capability of the application such as creating `Order`s. `Controller`s use the logic defined in `Core` and use injected dependencies (through inversion of control) to handle communication with external `Service`s.
+
+## Service
+
+A `Service` provides a dependency needed by a `Controller`.
```

Comment:
Services are more general. Controllers consume services for sure, but Services can be used anywhere.
Generally the type of Service dictates where it should be used.
An infrastructure service is used in Controllers, but not in domain methods etc.